src/linear_algebra/linear_independent.lean

```diff
@@ -297,6 +297,10 @@ begin
       exacts [λ _, zero_smul _ _, λ _ _ _, add_smul _ _ _] } }
 end
 
+lemma set.linear_dependent_iff {s : set ι} : ¬ linear_independent R (v ∘ coe : s → M) ↔
+  ∃ f : ι →₀ R, f ∈ finsupp.supported R R s ∧ ∑ i in f.support, f i • v i = 0 ∧ f ≠ 0 :=
+by simp [linear_independent_comp_subtype, finsupp.total_apply, finsupp.sum]
```

Comment:
This statement and name would be a lot more consistent with `linear_independent_comp_subtype` above
```suggestion
lemma linear_dependent_comp_subtype {s : set ι} :
  ¬linear_independent R (v ∘ coe : s → M) ↔
  ∃ f : ι →₀ R, f ∈ finsupp.supported R R s ∧ finsupp.total ι M R v f = 0 ∧ f ≠ 0 :=
by simp [linear_independent_comp_subtype]
```
obviously it's then up to the user to expand `finsupp.sum` and `finsupp.total`, but that seems to be how most of the lemmas work in this file at the moment.

Reply:
I don't care about the name but I think that `∑ i in f.support, f i • v i` is much more readable than `finsupp.total ι M R v f`. I wish everything could be written in that readable form.

Reply:
Maybe the best compromise is simply to have both the version with `finsupp.total` expanded and not expanded?

Reply:
Ok, I did that.